src/middlewared/middlewared/plugins/datastore/connection.py

```diff
@@ -27,6 +27,19 @@ class Config:
     engine = None
     connection = None
 
+    @private
+    def handle_constraint_violation(self, row):
+        self.logger.warning("Row %d in table %s violates foreign key constraint on table %s.",
+                            row.rowid, row.table, row.parent)
+
+        if row.table == "directoryservice_idmap_domain" and row.rowid <= 5 and row.parent == "system_certificate":
+            self.logger.warning("Removing certificate id for default idmap table entry.")
```

Comment:
Let's put a comment explaining why we short-circuit here. (i.e. the fallout of deleting an entry from idmap table)